File: 프로그래머스/0/120851. 숨어있는 숫자의 덧셈 （1）/숨어있는 숫자의 덧셈 （1）.java
class Solution {
    public int solution(String my_string) {
        int sum = 0;
        for(int i=0; i<my_string.length(); i++) {
            char ch = my_string.charAt(i);
            
            if(Character.isDigit(ch)) {
                sum += Character.getNumericValue(ch);
            }
        }
        
        return sum;
    }
}

File: 프로그래머스/0/120851. 숨어있는 숫자의 덧셈 （1）/숨어있는 숫자의 덧셈 （1）.js
function solution(my_string) {
    let sum = 0;
    
    for(let i=0; i<my_string.length; i++) {
        if(!isNaN(my_string[i])) {
            sum += parseInt(my_string[i]);
        }
    }
    
    return sum;
    
//     return Array.from(my_string)
//            .filter(char => !isNaN(parseInt(char)))   
//            .reduce((sum, curr) => sum + parseInt(curr), 0);
    
//     return [...my_string].reduce((sum, cur) => Number(cur) ? sum + +cur : sum, 0);
    
//     return my_string.replaceAll(/[^0-9]/g, '').split('')
//                     .reduce((sum, cur) => sum + +cur, 0);
}